### ISSUE TITLE
Specified argument was out of the range of valid values.

### ISSUE THREAD

**Issue by ebubekirtrkr** [microsoft/python-language-server#1313]
Issue Type: <b>Bug</b>

When I open any python file "Specified argument was out of the range of valid values."  notification occurs. Moreover, Python autocomplete can't work. 
Note: I tried to update and downgrade Vscode version. I have tried on both version 1_35 and 1_36

## Environment data

- VS Code version: Code 1.35.1 (c7d83e57cd18f18026a8162d042843bda1bcf21f, 2019-06-12T14:27:31.086Z)
- Extension version (available under the Extensions sidebar): 2019.6.22090
- OS and version: Linux x64 4.15.0-54-generic
- Python version (& distribution if applicable, e.g. Anaconda): Python 2.7.15+
- Type of virtual environment used (N/A | venv | virtualenv | conda | ...): N/A
- Relevant/affected Python packages and their versions: `clipboard==0.0.4`,` random` , `selenium==3.141.0`
- Jedi or Language Server? (i.e. what is `"python.jediEnabled"` set to; more info #3977): XXX

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Celeron(R) CPU 1007U @ 1.50GHz (2 x 945)|
|GPU Status|2d_canvas: enabled<br>checker_imaging: disabled_off<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: disabled_off<br>native_gpu_memory_buffers: disabled_software<br>rasterization: disabled_software<br>surface_synchronization: enabled_on<br>video_decode: unavailable_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|1, 1, 2|
|Memory (System)|3.74GB (0.11GB free)|
|Process Argv|--unity-launch|
|Screen Reader|no|
|VM|0%|
</details>

## Expected behaviour

XXX

## Actual behaviour

XXX

## Steps to reproduce:
1. It is a pop-up notification in VSCode.

### Pyhton Output Console

```
Starting Microsoft Python language server.
[Info  - 12:49:58 AM] Analysis cache path: /home/ebubekirtrkr/.cache/Microsoft/Python Language Server
[Info  - 12:49:58 AM] GetCurrentSearchPaths /usr/bin/python 
[Info  - 12:49:59 AM] Interpreter search paths:
[Info  - 12:49:59 AM]     /usr/lib/python2.7
[Info  - 12:49:59 AM]     /usr/lib/python2.7/plat-x86_64-linux-gnu
[Info  - 12:49:59 AM]     /usr/lib/python2.7/lib-tk
[Info  - 12:49:59 AM]     /usr/lib/python2.7/lib-dynload
[Info  - 12:49:59 AM]     /home/ebubekirtrkr/.local/lib/python2.7/site-packages
[Info  - 12:49:59 AM]     /usr/local/lib/python2.7/dist-packages
[Info  - 12:49:59 AM]     /usr/lib/python2.7/dist-packages
[Info  - 12:49:59 AM]     /usr/lib/python2.7/dist-packages/gtk-2.0
[Info  - 12:49:59 AM] User search paths:
[Info  - 12:49:59 AM]    /home/ebubekirtrkr/Desktop/Codes/SeleniumScripts/~/google-cloud-sdk/platform/google_appengine/lib/webapp2-2.5.2
[Info  - 12:49:59 AM]     /home/ebubekirtrkr/Desktop/Codes/SeleniumScripts/~/google-cloud-sdk/platform/google_appengine
[Info  - 12:49:59 AM]     /home/ebubekirtrkr/Desktop/Codes/SeleniumScripts/~/google-cloud-sdk/lib
[Info  - 12:49:59 AM]     /home/ebubekirtrkr/Desktop/Codes/SeleniumScripts/~/google-cloud-sdk/platform/google_appengine/lib/endpoints-1.0
[Info  - 12:49:59 AM]     /home/ebubekirtrkr/Desktop/Codes/SeleniumScripts/~/google-cloud-sdk/platform/google_appengine/lib/protorpc-1.0
[Error - 12:50:00 AM] Server initialization failed.
  Message: Specified argument was out of the range of valid values.
  Code: -32000 
   at System.ReadOnlySpan`1..ctor(Void* pointer, Int32 length)
   at Interop.Sys.DirectoryEntry.GetName(Span`1 buffer)
   at System.IO.Enumeration.FileSystemEntry.get_FileName()
   at System.IO.Enumeration.FileSystemEnumerableFactory.<>c__DisplayClass3_0.<UserFiles>b__1(FileSystemEntry& entry)
   at System.IO.Enumeration.FileSystemEnumerable`1.DelegateEnumerator.ShouldIncludeEntry(FileSystemEntry& entry)
   at System.IO.Enumeration.FileSystemEnumerator`1.MoveNext()
   at System.Collections.Generic.LargeArrayBuilder`1.AddRange(IEnumerable`1 items)
   at System.Collections.Generic.EnumerableHelpers.ToArray[T](IEnumerable`1 source)
   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   at Microsoft.Python.Core.IO.PathUtils.EnumerateFiles(String root, String pattern, Boolean recurse, Boolean fullPaths)+MoveNext() in E:\A\_work\1\s\src\Core\Impl\IO\PathUtils.cs:line 289
   at System.Linq.Enumerable.SelectManySingleSelectorIterator`2.MoveNext()
   at Microsoft.Python.Analysis.Modules.Resolution.ModuleResolutionBase.ReloadModulePaths(IEnumerable`1& rootPaths) in E:\A\_work\1\s\src\Analysis\Ast\Impl\Modules\Resolution\ModuleResolutionBase.cs:line 120
   at Microsoft.Python.Analysis.Modules.Resolution.MainModuleResolution.ReloadAsync(CancellationToken cancellationToken) in E:\A\_work\1\s\src\Analysis\Ast\Impl\Modules\Resolution\MainModuleResolution.cs:line 237
   at Microsoft.Python.Analysis.Modules.Resolution.MainModuleResolution.InitializeAsync(CancellationToken cancellationToken) in E:\A\_work\1\s\src\Analysis\Ast\Impl\Modules\Resolution\MainModuleResolution.cs:line 62
   at Microsoft.Python.Analysis.Analyzer.PythonInterpreter.LoadBuiltinTypesAsync(String root, IServiceManager sm, CancellationToken cancellationToken) in E:\A\_work\1\s\src\Analysis\Ast\Impl\Analyzer\PythonInterpreter.cs:line 59
   at Microsoft.Python.Analysis.Analyzer.PythonInterpreter.CreateAsync(InterpreterConfiguration configuration, String root, IServiceManager sm, CancellationToken cancellationToken) in E:\A\_work\1\s\src\Analysis\Ast\Impl\Analyzer\PythonInterpreter.cs:line 67
   at Microsoft.Python.LanguageServer.Implementation.Server.InitializeAsync(InitializeParams params, CancellationToken cancellationToken) in E:\A\_work\1\s\src\LanguageServer\Impl\Implementation\Server.cs:line 145
   at Microsoft.Python.LanguageServer.Implementation.LanguageServer.Initialize(JToken token, CancellationToken cancellationToken) in E:\A\_work\1\s\src\LanguageServer\Impl\LanguageServer.Lifetime.cs:line 36

```

### Problems
```
No problems have been detected in the workspace so far.
```

**Comment by jakebailey** [comment]
How did `{workspaceFolder}` make it to us without having a replacement done by the extension or VS Code? `{workspaceFolder}/~/` itself is very strange. @ebubekirtrkr Have you modified extraPaths?

**Comment by ebubekirtrkr** [comment]
@jakebailey I have placed it in there, because I don't wanna share my directory path.

**Comment by jakebailey** [comment]
Oh, I see. That's nearly the same syntax as what's in the config.

Note that I don't think putting `~` in the config is going to work; `~` is typically expanded by a shell, but not in other cases. If you could share what your extraPaths is (even if it's partially redacted), then I'd be happy to give you an alternate method to try.

**Comment by jakebailey** [comment]
Actually, in context, `/~/` doesn't quite make sense either.

**Comment by ebubekirtrkr** [comment]
@jakebailey I see what you want to say and I edit it, it is my mistake, they(~) were at the beginning and I forgot to remove them.

**Comment by ebubekirtrkr** [comment]
@jakebailey I edit it as what it was at beginning.

**Comment by jakebailey** [comment]
We had some recent fixes around paths and Python 2.7. I'd appreciate it if you could try out our daily build by setting the following and reloading:

```json
"python.analysis.downloadChannel": "daily"
```

**Comment by ebubekirtrkr** [comment]
@jakebailey When I copy and paste it in settings , a notification occurs in problems section which is  ` Unknown configuration setting`. Also, in VSCode settings there is no `python.analysis.downloadChannel `option.
What I need to do?

**Comment by jakebailey** [comment]
You can safely ignore the message. It's undocumented, and we give it out to help people test things.

**Comment by ebubekirtrkr** [comment]
@jakebailey  I have added what you say in settings and I reloaded VSCode. But It seems unsolved.

### Pyhton Output Console
```
Starting Microsoft Python language server.
Downloading https://pvsc.azureedge.net/python-language-server-daily/Python-Language-Server-linux-x64.0.3.29.nupkg... complete
Unpacking archive... done
[Info  - 1:57:05 AM] Analysis cache path: /home/ebubekirtrkr/.cache/Microsoft/Python Language Server
[Info  - 1:57:05 AM] GetCurrentSearchPaths /usr/bin/python 
[Info  - 1:57:05 AM] Interpreter search paths:
[Info  - 1:57:05 AM]     /usr/lib/python2.7
[Info  - 1:57:05 AM]     /usr/lib/python2.7/plat-x86_64-linux-gnu
[Info  - 1:57:05 AM]     /usr/lib/python2.7/lib-tk
[Info  - 1:57:05 AM]     /usr/lib/python2.7/lib-dynload
[Info  - 1:57:05 AM]     /home/ebubekirtrkr/.local/lib/python2.7/site-packages
[Info  - 1:57:05 AM]     /usr/local/lib/python2.7/dist-packages
[Info  - 1:57:05 AM]     /usr/lib/python2.7/dist-packages
[Info  - 1:57:05 AM]     /usr/lib/python2.7/dist-packages/gtk-2.0
[Info  - 1:57:05 AM] User search paths:
[Info  - 1:57:05 AM]     /home/ebubekirtrkr/Desktop/Codes/SeleniumScripts/~/google-cloud-sdk/platform/google_appengine/lib/webapp2-2.5.2
[Info  - 1:57:05 AM]     /home/ebubekirtrkr/Desktop/Codes/SeleniumScripts/~/google-cloud-sdk/platform/google_appengine
[Info  - 1:57:05 AM]     /home/ebubekirtrkr/Desktop/Codes/SeleniumScripts/~/google-cloud-sdk/lib
[Info  - 1:57:05 AM]     /home/ebubekirtrkr/Desktop/Codes/SeleniumScripts/~/google-cloud-sdk/platform/google_appengine/lib/endpoints-1.0
[Info  - 1:57:05 AM]     /home/ebubekirtrkr/Desktop/Codes/SeleniumScripts/~/google-cloud-sdk/platform/google_appengine/lib/protorpc-1.0
[Error - 1:57:05 AM] Server initialization failed.
  Message: Specified argument was out of the range of valid values.
  Code: -32000 
   at System.ReadOnlySpan`1..ctor(Void* pointer, Int32 length)
   at Interop.Sys.DirectoryEntry.GetName(Span`1 buffer)
   at System.IO.Enumeration.FileSystemEntry.get_FileName()
   at System.IO.Enumeration.FileSystemEnumerableFactory.<>c__DisplayClass3_0.<UserFiles>b__1(FileSystemEntry& entry)
   at System.IO.Enumeration.FileSystemEnumerable`1.DelegateEnumerator.ShouldIncludeEntry(FileSystemEntry& entry)
   at System.IO.Enumeration.FileSystemEnumerator`1.MoveNext()
   at System.Collections.Generic.LargeArrayBuilder`1.AddRange(IEnumerable`1 items)
   at System.Collections.Generic.EnumerableHelpers.ToArray[T](IEnumerable`1 source)
   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   at Microsoft.Python.Core.IO.PathUtils.EnumerateFiles(String root, String pattern, Boolean recurse, Boolean fullPaths)+MoveNext() in E:\A\_work\2\s\src\Core\Impl\IO\PathUtils.cs:line 296
   at System.Linq.Enumerable.SelectManySingleSelectorIterator`2.MoveNext()
   at Microsoft.Python.Analysis.Modules.Resolution.ModuleResolutionBase.ReloadModulePaths(IEnumerable`1& rootPaths) in E:\A\_work\2\s\src\Analysis\Ast\Impl\Modules\Resolution\ModuleResolutionBase.cs:line 120
   at Microsoft.Python.Analysis.Modules.Resolution.MainModuleResolution.ReloadAsync(CancellationToken cancellationToken) in E:\A\_work\2\s\src\Analysis\Ast\Impl\Modules\Resolution\MainModuleResolution.cs:line 210
   at Microsoft.Python.Analysis.Modules.Resolution.MainModuleResolution.InitializeAsync(CancellationToken cancellationToken) in E:\A\_work\2\s\src\Analysis\Ast\Impl\Modules\Resolution\MainModuleResolution.cs:line 61
   at Microsoft.Python.Analysis.Analyzer.PythonInterpreter.LoadBuiltinTypesAsync(String root, IServiceManager sm, CancellationToken cancellationToken) in E:\A\_work\2\s\src\Analysis\Ast\Impl\Analyzer\PythonInterpreter.cs:line 59
   at Microsoft.Python.Analysis.Analyzer.PythonInterpreter.CreateAsync(InterpreterConfiguration configuration, String root, IServiceManager sm, CancellationToken cancellationToken) in E:\A\_work\2\s\src\Analysis\Ast\Impl\Analyzer\PythonInterpreter.cs:line 67
   at Microsoft.Python.LanguageServer.Implementation.Server.InitializeAsync(InitializeParams params, CancellationToken cancellationToken) in E:\A\_work\2\s\src\LanguageServer\Impl\Implementation\Server.cs:line 135
   at Microsoft.Python.LanguageServer.Implementation.LanguageServer.Initialize(JToken token, CancellationToken cancellationToken) in E:\A\_work\2\s\src\LanguageServer\Impl\LanguageServer.Lifetime.cs:line 36
```

**Comment by jakebailey** [comment]
Those google-cloud-sdk paths, did you set up some configuration in order to add them? As in extraPaths or PYTHONPATH?

**Comment by ebubekirtrkr** [comment]
Yes in `settings.json` there is an option. 
```
"python.autoComplete.extraPaths": [
    "~/google-cloud-sdk/platform/google_appengine/lib/webapp2-2.5.2",
    "~/google-cloud-sdk/platform/google_appengine",
    "~/google-cloud-sdk/lib",
    "~/google-cloud-sdk/platform/google_appengine/lib/endpoints-1.0",
    "~/google-cloud-sdk/platform/google_appengine/lib/protorpc-1.0"
],
```

**Comment by jakebailey** [comment]
That's probably where things are going wrong. Nothing between those settings and our use of it expands the `~` into your real `$HOME`. (See also #557.)

What you can try is making those paths absolute. Based on your previous logs, I'd say something like:

```
"python.autoComplete.extraPaths": [
    "/home/ebubekirtrkr/google-cloud-sdk/platform/google_appengine/lib/webapp2-2.5.2",
    "/home/ebubekirtrkr/google-cloud-sdk/platform/google_appengine",
    "/home/ebubekirtrkr/google-cloud-sdk/lib",
    "/home/ebubekirtrkr/google-cloud-sdk/platform/google_appengine/lib/endpoints-1.0",
    "/home/ebubekirtrkr/google-cloud-sdk/platform/google_appengine/lib/protorpc-1.0"
],
```

Note that we have other issues when it comes to the ordering of these paths; some of those may not end up working due to the way we carve out parts of folders as import roots. You may need to reorder the above deepest-first, like:

```
"python.autoComplete.extraPaths": [
    "/home/ebubekirtrkr/google-cloud-sdk/platform/google_appengine/lib/endpoints-1.0",
    "/home/ebubekirtrkr/google-cloud-sdk/platform/google_appengine/lib/protorpc-1.0"
    "/home/ebubekirtrkr/google-cloud-sdk/platform/google_appengine/lib/webapp2-2.5.2",
    "/home/ebubekirtrkr/google-cloud-sdk/platform/google_appengine",
    "/home/ebubekirtrkr/google-cloud-sdk/lib",
],
```

**Comment by MikhailArkhipov** [comment]
The exception seems to happen deep in .NET framework. See https://github.com/dotnet/corefx/issues/34359. It is fixed in 3.0 while we are on 2.2.

**Comment by MikhailArkhipov** [comment]
We have upgraded LS to 3.0 a while ago in https://github.com/microsoft/python-language-server/pull/1521
